tests/test_maudau.py

```diff
@@ -0,0 +1,127 @@
+import datetime as DT
+import json
+import os
+import pytest
+import tempfile
+from pyspark.sql import SparkSession
+from pyspark.sql.types import StructField, StructType, StringType
+from mozetl.maudau import maudau as M
+
+NARROW_SCHEMA = StructType([
+    StructField("client_id",             StringType(),  True),
+    StructField("submission_date_s3",    StringType(),  False),
+    StructField("subsession_start_date", StringType(), True)])
+
+generated = M.format_as_submission_date(M.today.isoformat())
+
+spark = (
+    SparkSession
+    .builder
+    .master("local")
+    # Uncomment to run while offline: from pyspark.conf import SparkConf
+    # .config(conf=SparkConf().set("spark.driver.host", "localhost"))
+    .appName("maudau")
+    .getOrCreate()
+)
+
+
+@pytest.fixture
+def make_frame():
+    cols = ['client_id', 'subsession_start_date', 'submission_date_s3']
+    values = [
+        ('a', '2017-05-01', '20170508'),
+        ('b', '2017-05-01', '20170501'),
+        ('c', '2017-05-01', '20170510'),
+        ('a', '2017-05-02', '20170503'),
+        ('b', '2017-05-03', '20170503'),
+        ('b', '2017-05-04', '20170511'),
+        ('a', '2017-05-05', '20170507'),
+    ]
+    rows = map(json.dumps, [dict(zip(cols, tup)) for tup in values])
```

Comment:
Nit: I think you should be able to create a dataframe by using `spark.createDataFrame(list_of_dicts, schema)`, instead of going through a json -> rdd -> dataframe transformation.